checkov/terraform/checks/resource/base_resource_negative_value_check.py

```diff
@@ -0,0 +1,62 @@
+from abc import abstractmethod
+
+import dpath
+
+from checkov.common.models.enums import CheckResult
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class BaseResourceNegativeValueCheck(BaseResourceCheck):
+    def __init__(self, name, id, categories, supported_resources):
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        excluded_key = self.get_excluded_key()
+        if excluded_key is not None:
+            if dpath.search(conf, excluded_key) != {}:
+                value = dpath.get(conf, excluded_key)
+                if isinstance(value, list) and len(value) == 1:
+                    value = value[0]
+                if self.check_excluded_condition(value):
+                    return CheckResult.PASSED
+
+        inspected_key = self.get_inspected_key()
+        bad_values = self.get_bad_values()
+        if dpath.search(conf, inspected_key) != {}:
+            if len(bad_values) == 0:
```

Comment:
In what scenario would this condition would be true? If the check implements the negative-values check, it should have at least one forbidden value of the inspected key

Reply:
@tronxd There are two checks that forbid the key from existing at all, regardless of its value:
`RDSPubliclyAccessible (CKV_AWS_17)` https://github.com/bridgecrewio/checkov/pull/358/files#diff-1925ff1f8b443c99418c8f5e5efee48d
`GoogleComputeExternalIP (CKV_GCP_40)` https://github.com/bridgecrewio/checkov/pull/358/files#diff-8a9c56fa1e4a7208e776d2ae7716ece3

Reply:
@rotemavni If the attribute shouldn't exist at all, wouldn't it suffice to check the upper if? (`dpath.search(conf, inspected_key) != {}`)

Reply:
@tronxd I don't think it will suffice because when you check `if value in bad_values`, and `bad_values` is empty, the condition will be `False` and the check will pass instead of fail. 
I can merge these conditions into one if : `if value in bad_values or len(bad_values) == 0: return CheckResult.FAILED`

Reply:
@rotemavni agree. just rename the vars `bad_values` to `forbidden_values` :)